src/main/java/com/ryanharter/autoparcel/AutoParcelExtension.java

```diff
@@ -51,16 +48,20 @@ public boolean mustBeAtEnd(Context context) {
   }
 
   @Override
-  public String generateClass(final Context context, final String className, final String classToExtend, boolean isFinal) {
+  public String generateClass(final Context context, final String className,
+                              final String classToExtend, boolean isFinal) {
     return getParcelableCode(context, className, classToExtend, context.properties());
   }
 
   String getParcelableCode(Context context, String className, String classToExtend,
-                                   Map<String, ExecutableElement> properties) {
+                           Map<String, ExecutableElement> properties) {
+    validateProperties(context.processingEnvironment(), properties);
+
     Map<String, TypeName> types = convertPropertiesToTypes(properties);
 
     FieldSpec classLoader = FieldSpec
-        .builder(ClassName.get(ClassLoader.class), "CL", Modifier.PRIVATE, Modifier.FINAL, Modifier.STATIC)
+        .builder(ClassName.get(ClassLoader.class), "CL", Modifier.PRIVATE, Modifier.FINAL,
+            Modifier.STATIC)
         .initializer("$N.class.getClassLoader()", className)
```

Comment:
Unrelated: but I think you want `$T` here just in case there's a name clash of classes